caluma/caluma_core/filters.py

```diff
@@ -425,10 +427,19 @@ def filter(self, qs, value):
                     ),
                 )
                 lookup = {f"field_val__{lookup_expr}": expr["value"]}
+            elif lookup_expr == JSONLookupMode.INTERSECTS:
+                lookup = {}
+                exprs = [
+                    Q(**{f"{self.field_name}__{expr['key']}__contains": val})
+                    for val in expr["value"]
+                ]
+                # connect all expressions with OR
+                qs = qs.filter(reduce(lambda a, b: a | b, exprs))
```

Comment:
In this branch, you're updating `qs` directly. The other two branches set the `lookup` dict, and `qs.filter()` runs after.

You should probably move the `qs.filter()` call inside the branch in every case, so it's not called twice (See Line 443 below)